evolvekit/core/Ga/GaIsland.py

```diff
@@ -1,63 +1,283 @@
+            for offspring, crossover_indiv in zip(
+                self.offspring_population, bin_crossover_list
+            ):
+                offspring.bin_chrom = crossover_indiv.bin_chrom
+
+        mutation_offspring = []
+        if self.real_mutation:
+            mutation_offspring = self.real_mutation.perform(
+                GaOperatorArgs(self, self.selection.category())
+            )
+
+        if self.bin_mutation:
+            mutation_offspring = self.bin_mutation.perform(
+                GaOperatorArgs(self, self.selection.category())
+            )
+
+        for i in range(self.population_size):
+            if np.random.random() < self.mutation_prob:
+                self.offspring_population[i] = mutation_offspring[i]
+
+        if self.real_clamp_strategy != GaClampStrategy.NONE:
+            for indiv in self.offspring_population:
+                domain_count = len(self.evaluator.real_domain())
+                for i in range(domain_count):
+                    gene_value = indiv.real_chrom[i]
+                    domain = self.evaluator.real_domain()[i]
+                    lower, upper = domain
+                    if not lower <= gene_value <= upper:
+                        indiv.real_chrom[i] = get_clamp_strategy(
+                            self.real_clamp_strategy
+                        )(gene_value, domain)
+
+        indices = np.random.choice(
+            len(self.offspring_population),
+            size=len(self.elite_population),
+            replace=False,
+        )
+
+        for elite_index, offspring_index in enumerate(indices):
+            self.offspring_population[offspring_index] = self.elite_population[elite_index]
+
+        self.current_population = self.offspring_population
+        self.selected_population = []
+        self.offspring_population = []
+        self.elite_population = []
+
+    def __perform_crossover(self, crossover: GaOperator) -> List[GaIndividual]:
+        crossover_list = []
+        while len(crossover_list) < self.population_size:
+            if np.random.random() < self.crossover_prob:
+                crossover_list.extend(
+                    crossover.perform(GaOperatorArgs(self, crossover.category()))
+                )
+            else:
+                crossover_list.append(random.choice(self.selected_population))
+        crossover_list = crossover_list[: self.population_size]
+        return crossover_list
 
     def __finish(self) -> GaResults:
-        pass
+        self.inspector.finish(self.statistic_engine)
+        return GaResults(self.statistic_engine)
 
     def run(self) -> GaResults:
-        pass
+        self.__verify()
+        self.__initialize()
+
+        while True:
+            self.__evaluate()
+            self.statistic_engine.advance(self)
+            action = self.inspector.inspect(self.statistic_engine)
+            if action is GaAction.TERMINATE:
+                break
+            self.__evolve()
```

Comment:
After calling `self.statistic_engine.advance(self)`, please check if generation number from statistics is larger than `max_generations`. If so, break the loop.